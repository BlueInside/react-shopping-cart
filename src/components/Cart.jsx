import { useState } from 'react';
import ShopPage from './ShopPage';
import ShoppingCart from './ShoppingCart';
function Cart() {
  const [cartItems, setCartItems] = useState([]);

  function addToCart(product) {
    if (!product.quantity) product = { ...product, quantity: 1 };
    setCartItems((prevItems) => [...prevItems, product]);
  }

  function removeFromCart(productId) {
    setCartItems((prevItems) =>
      prevItems.filter((item) => item.id !== productId)
    );
  }

  function updateProductQuantity(productId, newQuantity) {
    setCartItems((prevProducts) =>
      prevProducts.map((product) =>
        product.id === productId
          ? { ...product, quantity: newQuantity }
          : product
      )
    );
  }
  return (
    <>
      <ShopPage addToCart={addToCart} />
      <ShoppingCart
        cartItems={cartItems}
        removeFromCart={removeFromCart}
        updateProductQuantity={updateProductQuantity}
      />
    </>
  );
}

export default Cart;
