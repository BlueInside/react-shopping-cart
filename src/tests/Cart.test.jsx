import { render } from '@testing-library/react';
import { describe, expect, it } from 'vitest';
import Cart from '../components/Cart';

describe('Cart component', () => {
  it('Renders Cart component correctly', () => {
    const { container } = render(<Cart />);
    expect(container).toMatchSnapshot();
  });
  it('');
});
