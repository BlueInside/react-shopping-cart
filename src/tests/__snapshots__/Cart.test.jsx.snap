// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Cart component > Renders Cart component correctly 1`] = `<div />`;
